src/helpers/getCountries.ts

```diff
@@ -1,7 +1,9 @@
 import { CountriesData } from "../types/CountriesTypes";
 
-export const getCountries = async () => {
-  const url = 'https://restcountries.com/v3.1/all';
+export const getCountries = async (endpoint?:string) => {
+  let url = `https://restcountries.com/v3.1/`;
```

Comment:
Puedes mandarle el endpoint directo:

```suggestion
  let url = `https://restcountries.com/v3.1/${endpoint || "all"}`;
```

Ya cuando se utilize puedes hacer algo como:

```tsx
await getCountries(`/name/${countryName}`)
```